### PR TITLE
Updates the objectc docs script to use env variables.

### PR DESCRIPTION
**Pull request by godofredoc**
This is in preparation to split the recipes into builder/tester. Adding
the env variable support will allow us to call different scripts from
configurations and coordinate artifacts location between generator steps
and archiving steps.

Bug: https://github.com/flutter/flutter/issues/84300

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.
- [X] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: tools/gen_objcdoc.sh [hunk 1]
```diff
@@ -12,10 +12,20 @@ if [[ ! -f "$FLUTTER_UMBRELLA_HEADER" ]]
...
-if [[ $# -eq 0 ]]
-  then
-      echo "Error: Argument specifying output directory required."
-      exit 1
+
+# If the script is running from within LUCI we use the LUCI_WORKDIR, if not we force the caller of the script
+# to pass an output directory as the first parameter.
+OUTPUT_DIR=""
+
+if [[ -z $LUCI_CI ]]; then
```
Comment (zanderso) [inline tools/gen_objcdoc.sh:20]:
```suggestion
if [[ -z "$LUCI_CI" ]]; then
```

File: tools/gen_objcdoc.sh [hunk 2]
```diff
@@ -12,10 +12,20 @@ if [[ ! -f "$FLUTTER_UMBRELLA_HEADER" ]]
...
+  if [[ $# -eq 0 ]]; then
+    echo "Error: Argument specifying output directory required."
+    exit 1
+  else
+    OUTPUT_DIR=$1
```
Comment (zanderso) [inline tools/gen_objcdoc.sh:25]:
```suggestion
    OUTPUT_DIR="$1"
```

**Review by zanderso — COMMENTED** [review]
This script is missing a `set -e` at the top.

**Comment by godofredoc** [comment]
> This script is missing a `set -e` at the top.

Added that line to the top.

**Review by zanderso — APPROVED** [review]

**Pull request merged**